src/org/labkey/test/components/glassLibrary/CollapsiblePanel.java

```diff
@@ -0,0 +1,119 @@
+package org.labkey.test.components.glassLibrary;
+
+import org.labkey.test.Locator;
+import org.labkey.test.WebDriverWrapper;
+import org.labkey.test.components.Component;
+import org.labkey.test.components.WebDriverComponent;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+import static org.labkey.test.WebDriverWrapper.WAIT_FOR_JAVASCRIPT;
+
+public class CollapsiblePanel extends WebDriverComponent<CollapsiblePanel.ElementCache>
+{
+    final WebElement _el;
+    final WebDriver _driver;
+
+    public CollapsiblePanel(WebElement element, WebDriver driver)
+    {
+        _el = element;
+        _driver = driver;
+    }
+
+    @Override
+    public WebElement getComponentElement()
+    {
+        return _el;
+    }
+
+    @Override
+    public WebDriver getDriver()
+    {
+        return _driver;
+    }
+
+    public CollapsiblePanel expand()
+    {
+        if (!isExpanded())
+            elementCache().expandCollapseToggle.click();
+        WebDriverWrapper.waitFor(()-> isExpanded(),
+                "the panel did not expand", 2000);
+        return this;
+    }
+
+    public CollapsiblePanel collapse()
+    {
+        if (isExpanded())
+            elementCache().expandCollapseToggle.click();
+        WebDriverWrapper.waitFor(()-> !isExpanded(),
+                "the panel did not collapse", 2000);
+        return this;
+    }
+
+    public boolean isExpanded()
+    {
+        return elementCache().waitForBody()
+                .getAttribute("class").equals("panel-collapse collapse in");
+    }
+
+    public WebElement body()
+    {
+        return elementCache().waitForBody();
+    }
+
+    @Override
+    protected ElementCache newElementCache()
+    {
+        return new ElementCache();
+    }
+
+
+    protected class ElementCache extends Component<?>.ElementCache
+    {
+        WebElement header = Locator.tagWithClass("div", "collapsible-panel-header")
+                .findWhenNeeded(this).withTimeout(2000);
+        WebElement expandCollapseToggle = Locator.tagWithClass("span", "pull-right")
+                .findWhenNeeded(header).withTimeout(2000);
+        private Locator.XPathLocator body = Locator.tagWithClass("div", "panel-collapse");
```

Comment:
I used the panel-collapse element (parent to the panel-body element) for 2 reasons: 
1) its class is used to signal its expand/collapse state
2) the use case I'm looking at puts provenance/runbuilder-specific classes in the panel-body div, I decided not to assume that it would exist in other usage

If we want to make that assumption, okay